32 feat room manager

allorak333 opened the pull request:
# room manager \#32
## Overview
db에 저장되지 않는 정보를 한 개의 인스턴스에 모아 둘 class(RoomManager)를 정의하였습니다.
추후 추가될 로직인 게임이 진행이나, 방을 조회할 때 사용할 때 RoomManager를 통해 이루어질 수 있습니다.
따라서 내부에 저장된 room의 정보를 room_id를 통해 검색 할 수 있고, 이에 관한 로그와 접속 목록, 게임 상태를 수정할 수 있도록 함수를 추가하였습니다.

create_room : 새로운 방 생성 (포트 생성 및 자식 프로세스 생성 로직 추가 예정)
join_room : 기존의 방에 접속
exit_room : 방장이 나간다면 방장을 업데이트하고, 방이 비어있다면 방 삭제가 이루어짐
save_match : room_id를 통해 저장된 game_id를 불러와 match 에 대한 로그 저장

<!---- Resolves: #(Isuue Number) -->
## PR Type

- [x] feat

## PR Checklist
PR이 다음 요구 사항을 충족하는지 확인하세요.

- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

uridonget committed:
#15 feat(LocalAuth): 회원가입 이메일 인증 추가

1. TLS 관련 설정 추가
    - 위치 : backend/transcendence/settings.py (154:155)
    - 장고가 nginx와 같은 프록시 서버 뒤에서 HTTPS 요청 여부를 인식하도록 돕는 설정.
    - 해당 장고 설정에 맞추어 프론트의 nginx 설정 파일 수정 필요.

2. Send Mail 관련 설정 추가
    - 위치 : backend/transcendence/settings.py (157:172)
    - 장고가 smtp를 사용해서 이메일을 전송할 때 필요한 설정.
    - 해당 기능을 사용하려면 TLS 설정이 필요해서 위에서 추가하였음.

3. CORS header 라이브러리 추가
    - 위치 : backend/requirements.txt (7:7)
    - 프론트에서 백엔드로 api 요청을 보낼 때 발생하는 CORS 요청을 허용하기 위한 라이브러리.

4. CORS header 관련 설정 추가
    - 위치 : backend/transcendence/settings.py (44:44) (68:68) (174:189)
    - 개발단계라서 모든 출처의 모든 메소드를 허용하도록 설정해두었음.
    - 추후 회의를 통해 CORS 관련 설정을 어떻게 설정할지 합의가 필요.
    - 해당 장고 설정에 맞추어 프론트의 nginx 설정 파일 수정 필요.

5. TLS 설정에 맞추어 함수 수정
    - 위치 : backend/account/views.py
    - (수정 전 : http://localhost) => (수정 전 : https://localhost)
    - JWT 발급 할 때 set_cookie의 설정을 False에서 True로 변경

6. LocalAuth 회원가입 과정에서 입력한 이메일로 인증메일을 발송하는 send_authentication_email 함수 추가
    - 위치 : backend/account/views.py
    - 인증코드를 이메일로 전송하도록 요청하는 함수 추가.

7. LocalAuth 회원가입 과정에서 인증코드를 제출하는 authenticate_code 함수 추가
    - 위치 : backend/account/views.py
    - 인증코드를 장고서버로 보내 발급한 인증코드와 비교하는 함수 추가.

8. local_auth_sign_up 함수 수정
    - 위치 : backend/account/views.py
    - 이메일 인증여부를 확인하고 유저를 생성하도록 수정.

uridonget committed:
#15 fix(env): 서버 URL을 환경변수에 추가

서버 URL(기존의 https://localhost)를 환경변수로 설정하여 변경이 용이하도록 수정.

uridonget committed:
#15 fix(env): 함수 누락 수정

os.environ 까지만 작성했던 누락된 코드를 수정.
(수정 전 : os.environ) => (수정 후 : os.environ.get)

uridonget committed:
#15 fix(CORS): CORS 관련 설정 오타 수정

장고 표준에 맞추어 허용되는 헤더 수정
(수정 전 : x-csrf-token) => (수정 후 : x-csrftoken)

uridonget committed:
Merge pull request #16 from exit-42/15-feat-make-email-authentication-for-localauth-signup

#15 feat(LocalAuth): 회원가입 이메일 인증 추가

uridonget committed:
#17(JWT): JWT 검증 및 갱신하는 함수 추가

authenticate_token 함수 추가
    - 사용자 쿠키의 access_token, refresh_token을 검증
    - 토큰 검증 성공/토큰 갱신 성공/토큰 검증 실패로 분기
앞으로 작성되는 API 요청 로직에 삽입하여 권한 검증에 사용
토큰 갱신 성공 시 프론트엔드에서 추가 처리 필요

uridonget committed:
#17(JWT): JWT를 통해 유저 데이터를 불러오는 함수 추가

login 함수 추가
    - JWT가 유효하면 user_data와 상태 코드 반환
    - JWT가 유효하지 않으면 에러 메시지와 상태 코드 반환
    - access_token 갱신 시 “유효하지 않은 것”으로 간주되어 별도 처리 필요
서버 상태를 변경하지 않으므로 GET 메소드 사용

uridonget committed:
#17(JWT): 쿠키의 JWT를 삭제하는 함수 추가

logout 함수 추가
    - JWT가 유효하면 유저의 쿠키에 저장된 JWT를 삭제하고 로그아웃 성공메시지 반환
    - JWT가 유효하지 않으면 에러 메시지와 상태 코드 반환
서버 상태를 변경하지 않으므로 GET 메소드 사용

uridonget committed:
#17(JWT): 인증 관련 에러 코드 수정

authenticate_token 함수에서 Access Token 갱신 후에도 401 Unauthorized를 반환하도록 수정
    - 기존에는 갱신 성공 시 200을 반환했으나, 토큰을 인증한다는 해당 함수의 목적에 맞게 사용되지 못하였기에 수정하였음

JWT에서 추출한 'user_id'와 동일한 id를 가진 유저가 존재하지 않는 경우 401 Unauthorized를 반환하도록 수정
    - 기존에는 404 Not Found를 반환했으나, 존재하지 않는 사용자를 공격자가 추론할 가능성을 고려하여 수정하였음

uridonget committed:
#17 Chore(message): Response 리턴값 수정

기존에는 4xx~5xx 상태코드를 가지면 응답에 error 필드를 넣고 2xx~3xx 상태코드를 가지면 message 필드를 넣었는데
이것을 message 필드로 통일하여 프론트에서 작업을 용이하게 할 수 있도록 수정하였음.

uridonget committed:
#17 Chore(response): Response 상태코드 수정

authenticate_token 함수에서 access_token을 refresh 하는 경우에는 상태코드 401이 아니라 452를 반환하도록 수정.

uridonget committed:
#17 Chore(HTTP method): login & logout method 수정

login 함수와 logout 함수가 POST method 요청으로 처리되도록 수정.

uridonget committed:
#17 Fix(logout): logout 로직 변경

기존에는 authenticate_token을 통해 토큰의 유효성을 검사하고나서 토큰을 삭제하는 로직을 실행하였는데
로그아웃 로직에서 토큰의 유효성을 검사할 필요가 없을 것 같다는 생각에 토큰 검증과정을 삭제하였습니다.

uridonget committed:
Merge pull request #18 from exit-42/17-feat-login-logout-feature-using-jwt

#17 feat(JWT) login logout feature

uridonget committed:
#19 Feat(Follow): Follow 모델 추가

1. models.py 파일에 Follow 클래스 추가.
2. admin.py 파일에 FollowAdmin 클래스 추가.

uridonget committed:
#19 Feat(Follow): 유저를 닉네임으로 탐색하는 함수 추가

search_user_by_nickname 함수를 추가하였음.

uridonget committed:
#19 Chore(Follow): 유저를 닉네임으로 탐색하는 함수 수정

1. search_user_by_nickname 함수를 수정하였음.
    * 기능 수정사항
    - 수정 전 : 특정 닉네임과 일치하는 닉네임을 가진 유저가 존재하는지 탐색.
    - 수정 후 : 특정 단어가 포함된 닉네임을 가진 유저가 존재하는지 탐색.

    * 코드 수정
    - 클래스형 뷰로 수정.

2. search_user_by_nickname 함수를 호출하는 urls endpoint 추가

uridonget committed:
#19 Fix(JWT): JWT 인증함수(authenticate_token) 수정

access_token이 소실되는 경우에 대한 처리를 추가하였음.

uridonget committed:
#19 Feat(CSRF): CSRF 토큰을 발급하는 함수 추가

- 프론트 서버에서 CSRF 토큰을 요청했을 때 토큰을 발급하는 get_csrf_token 함수를 추가하였음.
- CSRF 관련 설정들을 settings.py 파일에 추가하였음.

uridonget committed:
#19 Feat(CSRF): CSRF 토큰을 발급하는 함수 추가

- 프론트 서버에서 CSRF 토큰을 요청했을 때 토큰을 발급하는 get_csrf_token 함수를 추가하였음.
- CSRF 관련 설정들을 settings.py 파일에 추가하였음.

uridonget committed:
#19 Feat(Follow): 팔로우 함수 추가

- 특정 유저를 팔로우 하는 함수를 추가하였음. (Follow 클래스의 post method)
- @csrf_protect 데코레이터를 사용하여 csrf 검증을 추가하였음.

uridonget committed:
#19 Feat(Follow): 팔로우 목록을 불러오는 함수 추가

- accout/views.py 파일에 get_follows 함수를 추가하였음.
- 유저가 팔로우하고있는 모든 유저의 데이터를 JsonResponse 형태로 반환함.

uridonget committed:
#19 Feat(Follow): 팔로우를 취소하는 함수 추가

- 특정 유저를 언팔로우 하는 함수를 추가하였음. (Follow 클래스의 delete method)
- 팔로우 정보를 가지고 있는 Follow 클래스의 특정 객체를 삭제함.

uridonget committed:
#19 Fix(Follow): Follow 모델 수정

* 문제 상황
- Follow 모델에서 userAId와 userBId가 OneToOneField로 설정되어 있어
  하나의 유저가 단 한 명의 유저만 팔로우 할 수 있도록 제한되어 있음.

* 수정
- userAId와 userBId를 ForeignKey로 변경하여 한 유저가 여러 명을 팔로우 할 수 있도록 수정하였음.
- ForeignKey로 User 객체를 직접 가르키기에 userAId를 userA로 수정하였음.(마찬가지로 userBId -> userB)

uridonget committed:
#19 Chore(Follow): get_follows 함수 수정

- Follow 테이블을 순회해서 특정 유저의 팔로워들을 찾는 기존의 방식을 수정하였음
- Follow 테이블의 userA related name을 통해서 팔로워를 한 번에 불러올 수 있음.
- 마찬가지의 방법으로 나를 팔로우 하는 사람들의 목록도 불러올 수 있으나 기획상에서 필요가 없기에 추가하지 않았음.

uridonget committed:
merge-conflict fixed

uridonget committed:
merge-conflict fixed(2)

uridonget committed:
#19 Chore(Follow): PR feedback 수정

- 세션을 사용하지 않는 함수에서의 csrf 검사를 생략하였음.
- follow의 get 요청처리 함수에서 적절하지 않은 변수명을 수정하였음.
- settings.py에서 잘못된 설정을 수정하였음.

uridonget committed:
Merge pull request #20 from exit-42/19-feat-follow

#19 feat(Follow) follow feature

uridonget committed:
#21 Feat(2FA): 2FA 기능 추가

1. get_local_auth_token 함수의 로직을 반으로 나누고 그 사이에 2FA 인증 로직을 처리하였음.

2. signin_local_auth 함수 추가
    - 기존 get_local_auth_token 함수의 앞부분.
    - LocalAuth 아이디와 비밀번호를 parameter로 받아서 LocalAuth 테이블 내에 일치하는 유저 객체가 존재하는지 확인.
    - 아이디와 비밀번호가 일치하는 유저 객체를 발견한다면 세션에 2FA 코드를 저장하고 해당 유저 객체의 이메일로 코드를 전송.

3. get_local_auth_token 함수 수정
    - 기존 get_local_auth_token 함수의 뒷부분.
    - 2FA 코드를 parameter로 받아서 세션에 저장된 2FA 코드와 비교.
    - 입력받은 코드와 세션에 저장된 코드가 일치한다면 사용자의 쿠키에 JWT를 저장.

allorak333 committed:
Merge pull request #22 from exit-42/21-feat-2fa

#21 Feat(2FA): 2FA feature

uridonget committed:
#23 Feat(Profile): Nickname 변경 기능 추가

유저의 닉네임을 변경할 수 있는 change_nickname 함수를 추가하였다.
해당 함수는 PATCH 메소드의 요청을 처리한다.
새로운 닉네임은 해당 요청의 body에 'name' : new_nickname 형식으로 받아야한다.

uridonget committed:
#23 Feat(Profile): 유저 프로필 이미지 변경 기능 추가

1. docker-compose.yml 수정
    - 유저가 업로드 하는 이미지를 저장할 media 디렉토리를 nginx 디렉토리에 생성.
    - docker-compose 파일을 수정하여 해당 디렉토리와 볼륨을 연결.

2. settings.py 수정
    - media 디렉토리의 위치(컨테이너 내부)와 url을 설정.

3. change_profile_image 함수 추가
    - 유저가 업로드 한 이미지를 media/profile_images/ 디렉토리에 저장.
    - 파일 이름은 uuid로 생성된 username으로 저장.
    - 파일을 저장한 후, 데이터베이스의 User의 imagePath를 새로운 이미지 경로로 설정.

uridonget committed:
Merge pull request #25 from exit-42/23-feat-profile

#23 Feat(Profile): Profile management

uridonget committed:
#23 Refactor(Utils): 프로젝트 전체에서 사용할 util 파일 생성

core 디렉토리의 utils.py 파일을 생성하였음.

프로젝트 전체에서 사용할 함수들을 utils.py 파일에 작성할 것.

JWT 인증 함수인 authenticate_token 함수를 utils.py 파일로 이동하였음.

uridonget committed:
#23 Refactor(Follows): 팔로우 관리를 위한 앱을 생성

유저의 팔로우 관리를 위하여 follows 앱을 생성하였음.

Follows 앱으로 데이터베이스 Follow 모델을 이동하였음.

유저의 팔로우 관리에 사용되는 함수들을 follows 앱의 views.py 파일로 이동하였음.

관련 endpoint를 새로운 follow 앱에 맞추어 그대로 이동시키거나 소폭 수정하였음. (API 명세서에 수정사항 반영할 것)

uridonget committed:
#23 Chore(Follow): 앱 이름 수정

기존의 'follows' 앱 이름을 'follow' 으로 변경하였음.

기존의 'get_follows' 함수명을 'get_follower_list' 으로 변경하였음.

uridonget committed:
#23 Refactor(Authentication): 인증 앱 생성

기존의 'account' 앱에 포함되어 있었던 인증과정을 새로운 앱에 이동하였음.

authentication 앱 디렉토리 아래에 oauth 디렉토리와 local_auth 디렉토리를 생성하였음.

해당 디렉토리 아래에 views.py 파일과 urls.py 파일을 생성하여 해당되는 함수와 API 엔드포인트를 할당하여 코드의 가독성과 유지보수를 용이하게 하였음.

uridonget committed:
#23 Chore(Profile): 프로필 이미지 삭제 로직 수정

이전 이미지와 새로운 이미지의 확장자가 다를 때, 이전 이미지가 삭제되지 않는 오류를 발견하여 수정하였습니다.

uridonget committed:
#23 Chore(Profile): 프로필 이미지 저장 및 삭제 로직 수정

유저의 imagePath에 프로필 이미지 url을 저장할 때, 내부경로 뿐 아니라 프론트엔드 서버 url를 추가하도록 하였음.

위의 수정사항으로 인하여 기존의 프로필 이미지 삭제 로직을 수정하였음.

uridonget committed:
#23 Chore(Session): 세션 관련 함수에서 상태코드 수정

authenticate_code, get_local_auth_token 함수에서 '서버의 세션에 인증코드가 없는 경우'와 '잘못된 인증코드를 제출한 경우'에
동일한 상태코드(400)을 반환하는 것을 확인.

서버의 세션에 인증코드가 없는 경우에는 다시 이메일로 인증코드를 발송해야하고, 잘못된 인증코드를 제출한 경우에는 다시 입력값을 받아야하기에 상태코드를 다르게 설정하는 것이 유리함.

서버의 세션에 인증코드가 존재하지 않으면 404(Not found) 상태코드를 반환하도록 수정하였음.

allorak333 committed:
Merge pull request #28 from exit-42/27-refactoring-refactor-django-structure

#27 Refactor(App): Refactoring django structure

allorak333 committed:
26 docs api documentation (#29)

* chore(Setting) swagger 의존성 추가

* chore(Setting) swagger 의존성 추가

* chore(url setting) : swagger, redoc 페이지 추가

* docs(api documentation) : api에 swagger연결

swagger_auto_schema 데코레이터를 이용하여 api 문서화
api_view로 되어있지 않은 oauth는 제외한 나머지 api에 모두 데코레이터 추가

* refactor(code style) 수정된 파일에 black 적용

---------

Co-authored-by: Sangyun Han <sangyhan@cx2r6s7.42seoul.kr>

uridonget committed:
#30 Feat(Log): 게임 로그 모델 추가

게임로그 관리를 위한 Log 앱 추가하였음.
해당 앱의 모델 파일에 Game 클래스와 Match 클래스를 추가하였음.
Game 클래스와 Match 클래스를 admin 페이지에서 확인할 수 있도록 admin 파일을 수정하였음.

uridonget committed:
#30 Feat(Log): Game 등록 API 추가

새로운 게임을 등록하는 API를 추가하였음.

해당 API는 내부에서 사용되는 API로 게임을 시작할 때 사용하게 될 것.

POST 메소드로 처리되며 body에 type를 받음.

body의 type이 tournament이면 Game 모델의 isTournament를 True로 저장함.

uridonget committed:
#30 Feat(Log): Game 진행상태 변경 API 추가

Game의 isEnd를 True로 변경하는 API를 추가하였음.

해당 API는 서버 내부에서 사용되는 API로 게임이 끝났을 때 사용하게 될 것.

PATCH 메소드로 처리되며 body에 Game의 id(pk)를 받음.

uridonget committed:
#30 Feat(Log): Match 생성 API 추가

단일 경기(Match)가 끝난 후, 경기의 결과를 기록하는 Match를 생성하는 API를 추가하였음.

해당 API는 서버 내부에서 사용되는 API로 게임이 끝났을 때 사용하게 될 것.

POST 메소드로 처리되며 요청은 body에 게임의 결과를 저장하는데 필요한 인자들을 담아야 함.

uridonget committed:
#30 Chore(User): 유저 admin 수정

admin 페이지에서 User의 id(pk)도 확인할 수 있도록 수정하였음.

uridonget committed:
#30 Chore(Log): Game, Match 수정

Game의 isEnd가 True 일 때(이미 게임이 종료된 경우) 에러를 반환하도록 수정하였음.

create_match_log 함수를 호출하는 url Endpoint를 추가하였음.

Game과 Match model의 def __str__(self)를 문자열로 수정하여 에러를 해결하였음.

uridonget committed:
#30 Chore(Log): Match 모델 수정

Match 모델의 gameId를 game으로 변경하고 Game을 외래키로 하도록 변경하였음.

내부 API 방식을 사용하지 않기로 결정하였음. 지금까지 만들어둔 내부 API 처리 함수들은 개발과정에서 테스트용으로만 사용할 것.

uridonget committed:
#30 Feat(Log): Game 정보를 가져오는 기능 추가

Game 테이블에서 isEnd 값이 True 인 Game 객체들에 대한 정보를 불러오는 함수를 추가하였음.

GET method로 작동하며 쿼리 파라미터에 game_type (토너먼트 혹은 일반)과 cursor(기본값 -1)를 담아서 보내야 함.

Cursor based Pagination을 지원함. (프론트 사이드에서 next_cursor를 받아서 다음 요청을 수정하는 로직이 필요)

해당 기능은 아직 완성된 상태가 아니며 게임에 대한 정보를 기반으로 게임과 연결된 Match 정보를 불러오도록 수정할 것.

uridonget committed:
#30 Feat(Log): Game 정보와 연결된 Match 정보를 가져오는 기능 추가

기존의 게임 정보를 불러오는 함수를 수정하여 게임과 연결된 매치의 정보를 불러오도록 수정하였음.

uridonget committed:
#30 Chore(Log): 예외처리 추가

get_game_logs 함수에서 유효하지 않은 Cursor 값이 들어오면 400번 상태코드를 반환하도록 수정하였음.

uridonget committed:
Merge pull request #31 from exit-42/30-feat-game-log

#30 Feat(Log): Game Log

allorak333 committed:
feat(log) : log에 utils  모듈 추가

match와 game에 해당하는 db table 에 추가할 함수 정의
game table의 isEnd를 수정하는 함수 정의

allorak333 committed:
feat(room manager) : room mananger 추가

in-memory로 방 목록, 방 접속 목록/인원 상태를 관리하는 내부 변수와
room_id로 로그를 추가할 수 있는 함수 구현

allorak333 committed:
refactor(RoomManager) : black 적용

allorak333 committed:
refactor(log util) : black 적용

uridonget commented on backend/log/utils.py:
```diff
@@ -0,0 +1,119 @@
+            game=game,
+            rank=rank,
+        )
+        return
```
성공과 실패가 동일한 결과를 반환하는 부분에서 에러처리가 제대로 되지 않을 수 있다는 생각이 듭니다. 간단하게 bool 타입으로라도 성공과 실패를 구분하는 것이 어떨까요?

allorak333 replied:
코드 작성 당시 오류 상황에 별도로 처리할 수 있는 부분이 없어 아예 그 부분을 생각하지 못했습니다. 그러나 로그라도 남기는 편이 추후 디버깅 과정에 유용할 것 같다는 생각이 드네요. 로그와 함께 추가하겠습니다!

uridonget commented on backend/log/utils.py:
```diff
@@ -0,0 +1,119 @@
+from log.models import Game, Match
+
+
+def create_game(game_type):
```
굳이 utils.py 가 아니라 그냥 log의 views.py 파일에 작성해도 괜찮을 것 같습니다. 파일의 구분에 의미가 있는지 잘 모르겠네요.

allorak333 replied:
사실 MVT 패턴에서 V에 해당하는 것이 views.py이기 때문에 views.py에도 넣을 수 있습니다.
하지만 http api가 아닌 내부 함수는 room_manager라는 다른 클래스에서 사용할 함수이기 때문에 혼동되지 않도록 의도적으로 분리 했었습니다. 이 부분에 대해서는 어떻게 생각하시나요?

uridonget commented on backend/log/utils.py:
```diff
@@ -0,0 +1,119 @@
+        game.isEnd = True
+        game.save()
+
+        return
```
성공과 실패가 동일한 결과(리턴값 없음)를 반환하는 부분을 간단하게라도 구분하면 좋을 것 같습니다.

allorak333 replied:
이 부분도 위와 동일하게 로그 기능과 성공 여부 리턴 추가하겠습니다.

uridonget commented on backend/log/utils.py:
```diff
@@ -0,0 +1,119 @@
+    """
+    try:
+        is_tournament = False
+        if game_type == "tournament":
```
굳이 입력값을 문자열("tournament")로 받을 필요는 없을 것 같습니다. 그냥 토너먼트인지 아닌지 bool 타입으로 받는 것이 사소하지만 더 좋을 것 같습니다.

allorak333 replied:
말씀해주신 대로 두 가지 유형만 있기에 bool이 더 간편하긴 합니다.
하지만 데이터 베이스에도 다른 유형 게임과의 호환성을 염두해서 설계를 진행했었고, 추후 게임 유형이 추가된다면 함수의 인자 유형을 바꿔야 할 상황이 발생하게 됩니다. 따라서 game_type은 문자열로 두는 것이 더 좋을 것 같습니다.

uridonget commented on backend/room/rooms.py:
```diff
@@ -0,0 +1,129 @@
+        room = self.rooms.get(room_id)
+        if room and player_name not in room["players"]:
+            room["players"].append(player_name)
+            room["player_number"] += 1
```
room 의 mode에 따라서 입장할 수 있는 인원에 제한을 거는 것이 좋을 것 같습니다.

allorak333 replied:
이 부분 또한 오류 처리가 미흡했습니다. 수정하겠습니다 👍

uridonget commented on backend/room/rooms.py:
```diff
@@ -0,0 +1,129 @@
+        result["gamd_id"] = room["game_id"]
+        create_match_log(result)
+        if room["mode"] == "tournament":
+            if result["rank"] == 2:
```
"1대1" 및 "토너먼트 1차전" 의 rank 단계와 "토너먼트 결승전"의 rank에 대한 정확한 설명이 주석으로 있으면 좋을 것 같습니다.

allorak333 replied:
해당 내용 추가하겠습니다!

allorak333 committed:
chore(log) : 오류 상황 시 로그 기능 추가

bool 반환으로 변경하여 외부에서 오류 상황을 감지할 수 있게 하였습니다.
내부에서도 디버깅에 용이하게 오류 상황 시 로그를 남기게 하였습니다.

allorak333 committed:
feat(roommanager) : 방 입장 인증 추가

자식 프로세스에 무분별한 연결 요청을 막기 위해 RoomManager에서 입장이 허용된 유저를 저장하고
해당 유저만 연결할 수 있도록 join_room과 accpet_room으로 분할 하였습니다.
django app 이름이 room이 될 것을 감안하여 모듈 이름을 변경하였습니다.
docker compose 로 열린 port만 사용하기 위해 get_open_port에서 환경변수로 범위를 제한했습니다.

allorak333 committed:
feat(room) : 방 정원 확인 추가

정원이 찰 경우 방 상태를 업데이트 하는 기능을 추가했습니다.
소켓 생성 방식을 pool 방식으로 변경하여 비효율적인 포트 연결 시도를
줄였습니다.

allorak333 committed:
refactor(black) : black 적용

uridonget reviewed (dismissed):
LGTM! 복잡한 로직인데 가독성이 참 좋네요!

uridonget reviewed (approved):
LGTM! 복잡한 로직인데 가독성이 참 좋네요!

uridonget reviewed (dismissed):
LGTM! 복잡한 로직인데 가독성이 참 좋네요!

HeonSeok98 reviewed (dismissed):
LGTM
